src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -3714,7 +3714,11 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
                   // useful when a default doesn't match parameter type, e.g. def f[T](x:T="a"); f[Int]()
                   checkNotMacro()
                   context.set(ContextMode.DiagUsedDefaults)
-                  doTypedApply(tree, if (blockIsEmpty) fun else fun1, allArgs, mode, pt)
+                  def checkRecursive(res: Tree): Unit =
+                    if (settings.warnRecurseWithDefault && !res.isErroneous && context.owner.hasTransOwner(funSym))
+                      context.warning(res.pos, "Recursive call used default arguments.")
```

Comment:
You could be a bit more helpful by doing something like `recursive call uses default arguments rather than the function's arguments`

Reply:
Right, because you think, "As opposed to what?" How about a, "Did you mean...?"  "Did you intend to thread the function's arguments down the stack?" Or only under `-Ysecond-guess`?

Reply:
Update the string from retronym minimalism.